templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -512,12 +569,20 @@ contents:
       local counter=0
       local iface_default_hint_file="$1"
       local extra_bridge_file="$2"
+      local ip_hint_file="$3"
       local extra_bridge=""
 
       if [ -f "${extra_bridge_file}" ]; then
         extra_bridge=$(cat ${extra_bridge_file})
       fi
 
+      # if node ip was set, we should search for interface that matches it
+      iface=$(get_interface_from_ip_hint "${ip_hint_file}" "${extra_bridge}")
+      if [[ -n "${iface}" ]]; then
+        echo "${iface}"
+        return
+      fi
```

Comment:
With this, the function does no longer behave as described by the function doc. Please update the doc to accurately describe how the function behaves and take into account how it affects the overall behavior.

This aspect is important:
```
If the default interface is br-ex, use that and return
```

How do you expect this script to behave if run manually after br-ex is already configured? Because currently that would be a noop but with this change, there would be a reconfiguration (with the subsequent network disruptions).

Reply:
actually nothing should happened, function will return br-ex and code on 
https://github.com/openshift/machine-config-operator/blob/master/templates/common/_base/files/configure-ovs-network.yaml#L699
will not run. Actually this should 100% same behavior as now. What i miss?

Reply:
Ah, I see now. 
Could you update the doc though mentioning the interface that holds the ip address hint will be selected over the default interface.
Looks like this method should rather be called now `get_nodeip_interface` and documented as obtaining the nodeip selected interface and fall-backing to the default interface.

Reply:
can you point me to the docs?

Reply:
I was just referring to the `get_default_interface` function comment doc:
```
    # Accepts parameters $iface, $iface_default_hint_file
    # Finds the default interface. If the default interface is br-ex, use that and return.
    # Never use the interface that is provided inside extra_bridge_file for br-ex1.
    # Never use br-ex1.
    # If the default interface is not br-ex:
    # Check if there is a valid hint inside iface_default_hint_file. If so, use that hint.
    # If there is no valid hint, use the default interface that we found during the step
    # earlier. Write the default interface to the hint file.
```